src/features/navbar/components/VerticalNav.tsx

```diff
@@ -38,6 +41,21 @@ export const VerticalNav: FC<Props> = ({ closeVerticalNav }) => {
 
   const defaultLanguageOption = __languageOptions.find(language => language.value === i18n.languages[0]);
 
+  const navCategoryMap = {
+    General: navLinks.filter(link => link.label === 'Directory'),
+    Management: navLinks.filter(link => link.label === 'Users'),
+  };
+
+  const getDefaultCategoryBasedOnLocation = () => {
+    if (location.pathname === '/users') {
+      return 'Management';
+    }
+    if (location.pathname === '/agents') {
+      return 'General';
+    }
+    return null;
+  };
```

Comment:
This is not very maintainable. Whenever someone adds a new navlink they need to reach in here and make sure to update these accordingly. The `navCategoryMap` in particular would be difficult to keep updated since there is no direct relationship between a category and a navLink.

Instead what do you think of removing the abstraction of `navLinks` entirely. Lets just declarative define them in the JSX.

An API like this looks like it would solve all the problems, while being easy to maintain.

![image](https://user-images.githubusercontent.com/2287977/187499064-b5c7296c-1b9c-4bee-aab6-7339b1c25960.png)

Reply:
Yeah. I agree. And, replacing the navLinks abstraction with a declarative JSX approach makes sense to me. I'll go ahead and refactor the current implementation. 

Also, thanks for the API Design tip. I'll keep it in mind for future API design decisions.